### PR TITLE
Respect -use-ld= on Darwin

### PR DESCRIPTION
**Pull request by benlangmuir**
Use the custom ld path rather than the system 'ld'.

### CODE REVIEW

**Comment by benlangmuir** [comment]
@swift-ci please test

File: Tests/SwiftDriverTests/SwiftDriverTests.swift [hunk 1]
```diff
@@ -1244,6 +1244,19 @@ final class SwiftDriverTests: XCTestCase {
...
+      // -use-ld
+      var driver = try Driver(args: commonArgs + ["-use-ld=/bin/my-ld"], env: env)
+      let plannedJobs = try driver.planBuild()
+
+      XCTAssertEqual(3, plannedJobs.count)
+      XCTAssertFalse(plannedJobs.contains { $0.kind == .autolinkExtract })
+
+      let linkJob = plannedJobs[2]
+      XCTAssertEqual(linkJob.kind, .link)
+      XCTAssertEqual(linkJob.tool.name, "/bin/my-ld")
```
Comment (compnerd) [inline Tests/SwiftDriverTests/SwiftDriverTests.swift:1257]: Could you please add a test case for the suffix form as well?

Comment (benlangmuir) [reply Tests/SwiftDriverTests/SwiftDriverTests.swift:1257]: Sorry, what is the "suffix form" in this context?

Comment (compnerd) [reply Tests/SwiftDriverTests/SwiftDriverTests.swift:1257]: `-use-ld=lld` should use `ld64.lld`.

Comment (benlangmuir) [reply Tests/SwiftDriverTests/SwiftDriverTests.swift:1257]: @compnerd  Ohh, thanks. I didn't realize that `-[f]use-ld` was designed for choosing a "variant" like this and that the version taking a path was deprecated.  It turns out the thing I wanted is spelled `--ld-path` in clang, and it doesn't exist at all in swiftc.  I'll just close this for now.

**Pull request closed without merging**